init

Casso23 opened the pull request:
MVP ATM
Домашняя работа 5

Casso23 committed:
init

Casso23 committed:
удалена тестовая реализация, удалены обертки Long

RodinDenis commented on src/main/java/ru/sber/javaschool/Session.java:
```diff
@@ -0,0 +1,50 @@
+@Data
+@Accessors(chain = true)
+public class Session {
+    public enum Status{NO_AUTHORIZATION,OPEN,CLOSE}
```
я предпочитаю перечисления никогда не описывать в классе даже если в текущей реализации оно используется только в этом классе. Это придает больше гибкости в плане использования.

RodinDenis replied:
дочитал до конца. Я предложил бы такие имена UNAUTHORIZED, OPENED, CLOSED

RodinDenis commented on src/main/java/ru/sber/javaschool/Session.java:
```diff
@@ -0,0 +1,50 @@
+    public enum Status{NO_AUTHORIZATION,OPEN,CLOSE}
+
+    private long cardNumber;
+    private Date startTime;
```
Не рекомендую в новом коде использовать старый API Date. Если требуется дата без времени и зоны, то можно воспользоваться LocalDate

RodinDenis commented on src/main/java/ru/sber/javaschool/Session.java:
```diff
@@ -0,0 +1,50 @@
+    private Date startTime;
+    private Date endTime;
+    private List<Operation> operations = new ArrayList<>();
+    private Status sessionStatus = Status.NO_AUTHORIZATION;
```
я бы назвал UNAUTHORIZED а не NO_AUTHORIZATION

RodinDenis commented on src/main/java/ru/sber/javaschool/Session.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    private void start(String pin){
+        OperationAuthorization operationAuthorization = new OperationAuthorization(this.cardNumber,pin);
```
если в рамках одного метода не возникает не однозначности между переменными с одинаковым именем, то использовать this настоятельно не рекомендую

RodinDenis commented on src/main/java/ru/sber/javaschool/customer/Customer.java:
```diff
@@ -0,0 +1,13 @@
+@Data
+@Accessors(chain = true)
+public class Customer {
+    private long id;
```
важно помнить что гораздо эффективнее работа будет с int. Использовать long нужно только в тех кейсах где это будет обосновано

Casso23 replied:
В данном случае long выбран осознано, предполагаю что ID владельца будет либо в long либо guid

RodinDenis commented on src/main/java/ru/sber/javaschool/processing/operations/Authorization.java:
```diff
@@ -0,0 +1,7 @@
+package ru.sber.javaschool.processing.operations;
+
+public interface Authorization {
+    enum Status {CARD_BLOCKED, INCORRECT_PIN, OK}
```
в данном контексте ok режет слух. Тут скорее всего подойдет success

RodinDenis commented on src/main/java/ru/sber/javaschool/processing/operations/OperationAuthorization.java:
```diff
@@ -0,0 +1,11 @@
+package ru.sber.javaschool.processing.operations;
+
+public class OperationAuthorization extends Operation{
```
имя суппер класса в имени саб класса обычно находится правее т.е. используется как суффикс

RodinDenis commented on src/main/java/ru/sber/javaschool/processing/operations/OperationAuthorization.java:
```diff
@@ -0,0 +1,11 @@
+    }
+
+    public boolean isAuthorization(){
+        return this.getStatus().equals(Status.OK);
```
enum принято сравнивать через ==. Это гарантирует что не будет NPE и самое главное проверка совместимости типов будет выполнена на этапе компиляции.

Casso23 committed:
enum вынесены из класса
Исправлены названия статусов
API Date заменено на LocalDateTime
Удалены this, где не возникает не однозначности
Исправлены имена саб классов в формате nameSuper
Сравнение enum реализовано через ==
Файлы сессии вынесены в отдельный пакет

Casso23 commented:
enum вынесены из класса
Исправлены названия статусов
API Date заменено на LocalDateTime
Удалены this, где не возникает не однозначности
Исправлены имена саб классов в формате nameSuper
Сравнение enum реализовано через ==
Файлы сессии вынесены в отдельный пакет
